CriticalMass/IBConstructable.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  IBConstructable.swift
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 19.04.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import UIKit
+
+protocol IBConstructable {
+    static var nibName: String { get }
+    static var bundle: Bundle { get }
+    static var nib: UINib { get }
+}
+
+extension IBConstructable where Self: NSObject {
+    static var nibName: String {
+        return typeName
+    }
+
+    static var bundle: Bundle {
+        return Bundle(for: Self.self)
+    }
+
+    static var nib: UINib {
+        return UINib(nibName: nibName, bundle: bundle)
+    }
+}
+
+extension IBConstructable where Self: UIViewController {
+    static func fromNib() -> Self {
+        return self.init(nibName: nibName, bundle: bundle)
+    }
+}
+
+extension IBConstructable where Self: UIView {
+    static func fromNib() -> Self {
```

Comment:
Then every View that wants to use the protocol needs to have a test. Throwing a fatalError is a good enough check IMO since then it would crash when running the app for a test

Reply:
Fair point. not the most important test case